src/main/java/org/apache/commons/io/DirectoryUtils.java

```diff
@@ -0,0 +1,149 @@
+            throws IOException {
+
+        boolean check = isEverythingInCompareDirectory(directory, compareDirectory, checkFileContent);
+
+        // we only need to check file content in on direction.
+        boolean checkOppositeFileContent = false;
+
+        boolean checkOpposite = check
+                && isEverythingInCompareDirectory(compareDirectory, directory, checkOppositeFileContent);
+        return check && checkOpposite;
+
+    }
+
+    /**
+     * checks if the directory file lists and file content is equal
+     * 
+     * @param directory
+     *            the directory
+     * @param compareDirectory
+     *            the directory to compare with
+     * @param checkFileContent
+     *            also compare file content
+     * @return true if directory and compareDirectory are equal
+     * @throws IOException
+     */
+    private static boolean isEverythingInCompareDirectory(Path directory, Path compareDirectory,
+            boolean checkFileContent) throws IOException {
+
+        if (directory != null && compareDirectory != null) {
+            // LOGGER.info("checking directory " + directory);
+            // LOGGER.info("checking compareDirectory " + compareDirectory);
+
+            File directoryFile = directory.toFile();
+            File compareFile = compareDirectory.toFile();
+
+            // check, if there is the same number of files/subdirectories
+            File[] directoryFiles = directoryFile.listFiles();
+            File[] compareFiles = compareFile.listFiles();
+
+            if (directoryFiles != null && compareFiles != null) {
+                //LOGGER.info("directoryFiles: " + directoryFiles.length + " vs compareFiles: " + compareFiles.length);
+                if (directoryFiles.length == compareFiles.length) {
+                    return compareDirectoryContents(directory, compareDirectory, checkFileContent);
+
+                } else {
+                    //LOGGER.info("number of files in directory are different " + directoryFiles.length + " vs compareDirectory: " + compareFiles.length);
+                    return false;
+                }
+            } else {
+                return checkForNulls(directoryFiles, compareFiles);
+            }
+
+        } else {
+            // one of the directories is null
+            return checkForNulls(directory, compareDirectory);
+        }
+
+
+    }
+
+    private static boolean checkForNulls(Object directory, Object compareDirectory) {
+        if (directory == null && compareDirectory == null) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private static boolean compareDirectoryContents(Path directory, Path compareDirectory, boolean checkFileContent)
+            throws IOException {
+
+        // LOGGER.info("compareDirectoryContents: " + directory);
+        try (DirectoryStream<Path> directoryStream = Files.newDirectoryStream(directory)) {
```

Comment:
How does this implementation account for file and directories present in `compareDirectory` but not in `directory`? Seems quite broken. You'd be better off with names like `directory1` and `directory2` IMO.